### ISSUE TITLE
Redis Key not unlocked after expiry time in v2.8.2101

### ISSUE THREAD

**Issue by RobertVuru** [microsoftarchive/redis#405]
Hi Team,

In our website , we have used redis v2.8.2101 and facing issues with set command along with NX parameter. below is the exact syntax which we implemented in our site. MaxMemory is set to 128MB and maxmemory-policy is set to "allkeys-lru" in redis configuration file.
As per our understanding , Key1 key will be created and locked for 60 sec on redis application. Key1 will be unlocked and expired after 60 sec.    

SET Key1 Value NX EX 60

Here the issue is key is not getting unlocked or expired after 60 sec. this is happening intermittently and we could not find any logs related to this issue.

Please let us know how to resolve this issue and let us know if any defect is logged on this.
Thanks

**Comment by enricogior** [comment]
Hi @RobertVuru 
thank you for reporting the issue, we will look into it and let you know.

**Comment by enricogior** [comment]
@RobertVuru 
the command you are executing sets the value and the expiration time if and only if the key does not exits, are you sure that the intermittently failure is not caused by the fact that the key already exists?
To verify that I would suggest to remove the NX flag.
Thank you.

**Comment by RobertVuru** [comment]
Yes you are correct , it creates a new key if it does not exist. 
Our main intention is key should not be created until 60 sec if we try to create it also. Hence we have used NX flag.
Sometimes Key is not expiring intermittently with above mentioned command. our functionality is breaking due to this issue.

Thank you.

**Comment by enricogior** [comment]
@RobertVuru 
since 2.8.2101 several updates with important fixes have been released, not specific to the problem you described, but we are not actively debugging any issue on 2.8.2101, therefore please upgrade to 2.8.2400 and let us know if the problem still persist with 2.8.2400.
Thank you.
